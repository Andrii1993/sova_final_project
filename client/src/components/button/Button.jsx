import { Btn } from './styles';
import PropTypes from 'prop-types';

const Button = ({ backgroundColor, text, onClick }) => {

    return (
      <Btn data-testid='test' onClick={onClick} backgroundColor={backgroundColor}>
        {text}
      </Btn>
    )
}

Button.propTypes = {
  backgroundColor: PropTypes.string,
  text: PropTypes.string,
  onClick: PropTypes.func
};

Button.defaultProps = {
  backgroundColor: "black",
  text: "Ok",
  onClick: null
};

export default Button;
