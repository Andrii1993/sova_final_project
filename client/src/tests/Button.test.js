import Button from '../components/button/Button'
import '@testing-library/jest-dom';
import {render, cleanup} from "@testing-library/react";
import renderer from 'react-test-renderer';

afterEach(cleanup);

it("renders with a text", () => {
  const{container}= render(<Button text="Button"/>);
  expect(container.textContent).toBe("Button");
});

it("renders with background", () => {
  const {getByTestId} = render(<Button backgroundColor='brown'/>);
  expect(getByTestId('test')).toHaveStyle('background-color: brown');
});


it('test SnapShot', () => {
  const buttonComp = renderer.create(<Button text="Button" backgroundColor="black"/>).toJSON()
  expect(buttonComp).toMatchSnapshot();
});

