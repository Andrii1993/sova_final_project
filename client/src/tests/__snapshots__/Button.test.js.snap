// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test SnapShot 1`] = `
<div
  className="sc-bczRLJ inrKHO"
  data-testid="test"
  onClick={null}
>
  Button
</div>
`;
